sci-libs/torchaudio/torchaudio-2.4.1.ebuild

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{11..12} )
+DISTUTILS_SINGLE_IMPL=1
+DISTUTILS_USE_PEP517=setuptools
+DISTUTILS_EXT=1
+ROCM_SKIP_GLOBALS=1
+inherit cuda distutils-r1 multiprocessing rocm
+
+DESCRIPTION="Data manipulation and transformation for audio signal processing"
+HOMEPAGE="https://github.com/pytorch/audio"
+SRC_URI="https://github.com/pytorch/audio/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/audio-${PV}"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="cuda rocm ffmpeg soundfile openmp"
+REQUIRED_USE="?? ( cuda rocm )"
+#RESTRICT="!test? ( test )"
+
+distutils_enable_tests pytest
+
+DEPEND="
+	ffmpeg? ( media-video/ffmpeg:= )
+	sci-libs/caffe2[cuda?,rocm?,openmp?]
+	sci-libs/pytorch[${PYTHON_SINGLE_USEDEP}]
+"
+RDEPEND="
+	${DEPEND}
+	$(python_gen_cond_dep '
+		soundfile? ( dev-python/soundfile[${PYTHON_USEDEP}] )
+	')
+"
+BDEPEND="
+	test? (
+		$(python_gen_cond_dep '
+			dev-python/numpy[${PYTHON_USEDEP}]
+			dev-python/typing-extensions[${PYTHON_USEDEP}]
+			dev-python/parameterized[${PYTHON_USEDEP}]
+			dev-python/scipy[${PYTHON_USEDEP}]
+		')
+	)
+"
+
+src_compile()
+{
+	export MAX_JOBS="$(makeopts_jobs)" # Let ninja respect MAKEOPTS
+
+	use cuda && export USE_CUDA=1
+	use rocm && addpredict /dev/kfd && export USE_ROCM=1
+	use openmp && export USE_OPENMP=1
```

Comment:
I think these switches should also be set if the use flag is disabled, else they could accidentally be enabled.

Reply:
Is there a one-liner for this?

Reply:
I think this will work: `export USE_CUDA=$(usex cuda 1 0)`